add Change member role via API and OpenAPI links (#2250)
huggingface/hub-docs: docs/hub/organizations-security.md

See [Tokens Management](./enterprise-tokens-management)

## Change member role via API

You can change a member's **organization role** (Read / Contributor / Write / Admin) and, optionally, their roles in **resource groups** using the Hub API. The API updates **one member per request**. To change roles for multiple members, call the API in a loop (examples below).

**OpenAPI reference:** <a href="https://huggingface.co/spaces/huggingface/openapi#tag/orgs/PUT/api/organizations/&#123;name&#125;/members/&#123;username&#125;/role" rel="nofollow">PUT /api/organizations/&#123;name&#125;/members/&#123;username&#125;/role</a>

### Prerequisites

- Your organization must have a **subscription plan** (e.g. Team or Enterprise). The endpoint returns 402 otherwise.
- You must be authenticated as an organization member with **Write** (or Admin) permission on the organization.
- The target user must already be a **member** of the organization.

### Base URL and authentication

- **Base URL:** `https://huggingface.co` (or your Hub host if self-hosted).
- **Authentication:** Send your token in the request header:
  ```http
  Authorization: Bearer <your_access_token>
  ```

>   Create a token with at least **Write** access at [https://huggingface.co/settings/tokens](https://huggingface.co/settings/tokens).

Reviewer: ```suggestion
  Create a fine-grained token with the "Write access to organizations settings / member management" permission scoped to your org at [https://huggingface.co/settings/tokens](https://huggingface.co/settings/tokens).
```
safer/best practice

### Change member role endpoint

**Request**

```http
PUT /api/organizations/{org_name}/members/{username}/role
Authorization: Bearer <your_access_token>
Content-Type: application/json

{
  "role": "read",
  "resourceGroups": []
}
```

- **Path parameters**
  - `org_name`: Organization slug (e.g. `my-org`).
  - `username`: Hugging Face **username** of the member whose role you are changing.
- **Body**
  - `role` (required): The member's **organization-level** role. One of: `"read"`, `"contributor"`, `"write"`, or `"admin"`.
  - `resourceGroups` (optional): Array of resource group assignments for this user. Each item:
    - `id`: Resource group ID (24-character hex string; get IDs from the [resource groups list API](./security-resource-groups#list-resource-groups)).
    - `role`: Role in that resource group: `"read"`, `"contributor"`, `"write"`, or `"admin"`.
  - If you omit `resourceGroups` or pass `[]`, the user is removed from all resource groups. To only change org role and leave resource groups unchanged, pass their current resource group memberships (the body always sets both org role and resource group list).

> **Example (curl) – set org role to "read", no resource groups**

Reviewer: ```suggestion
**Example (curl) – set org role to "read", no resource groups (removes any the user was previously in)**
```

```bash
curl -s -X PUT \
  -H "Authorization: Bearer $HF_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{"role":"read","resourceGroups":[]}' \
  "https://huggingface.co/api/organizations/my-org/members/member1/role"
```

> **Example (curl) – set org role and resource group roles**

Reviewer: ```suggestion
**Example (curl) – set org role and resource group roles (overrides any current groups)**
```
